doc/concepts/pipeline-concepts/pipeline/join.md

```diff
@@ -0,0 +1,111 @@
+# Join
+
+A join is a special type of pipeline input that enables you
+to combine files that reside in separate Pachyderm repositories
+and match a particular naming pattern. The join operator must
+be used in combination with a glob pattern that reflects a
+specific naming convention.
+
+A Pachyderm join is similar to a database equi-join operation that
+combines columns from one or more tables into its own dataset
+if the column names match.
+
+Similarly, joins help you consolidate results with matching names from
+multiple repositories in a single dataset. For example, you collect
+data from multiple sources and need to consolidate the results
+that match a specific parameter. These results can be readings
+from a sensor, a geographical location, or something similar.
+
+When you configure a join input, you must specify a glob pattern
+that defines a capture group. A capture group is the group of
+files selected for further processing. In the glob parameter,
```

Comment:
> In the glob parameter,
a capturing glob must be enclosed in parenthesis. For example,
if you have a file named `test`, you could configure the
following globs for it:

In the glob parameter, a capture group is specified by parentheses. Capture groups are numbered from left to right and can also be nested within each other. You can read more about capture group conventions here (LINK TO UNIX DOCS). Below are a few examples using a file path: `dir123/file456.txt`

`dir123/file(456).txt`  --> capture group 1 = `456`
`(dir123)/file(456).txt`  --> capture group 1 = `123`. capture group 2 = `456`. 
`dir123/file(4(5)6).txt`  --> capture group 1 = `456`. capture group 2 = `5`. 

--- 
started writing this out and realized that this actually needs to get reorganized a bit. I'm happy to take a rewrite pass on it, but I'll get to it tomorrow. Submitting this for now just to not lose track